src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -753,22 +753,27 @@ public void Init()
                 _sequenceNumber = 0;
             }
 
-            // IsReady returns the current _sequenceNumber, which must be passed to StartAsyncOperation below.
-            public bool IsReady(SocketAsyncContext context, out int observedSequenceNumber)
+            // IsReady returns whether an operation can be executed immediately.
+            // observedSequenceNumber must be passed to StartAsyncOperation.
+            public bool IsReady(SocketAsyncContext context, out int? observedSequenceNumber)
```

Comment:
Can we keep `observedSequenceNumber` as a non-nullable int? It's only used when IsReady returns true, so we should be able to just return 0 or whatever when that's not the case. Seems like a non-nullable int would be slightly more efficient here.

Reply:
> it's only used when IsReady returns true

When `IsReady` returns `false`, the `null` is used to ensure that `StartAsyncOperation` will try the operation in case the queue becomes ready in the meanwhile. To not use a nullable int here, the best value to use is `_sequenceNumber - 1` (when returning `false`), because that will fail the sequence number equals check in `StartAsyncOperation`.